realm-management/libs/import-users.js

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable no-console */
+const ldap = require('ldapjs');
+const fs = require('fs-extra');
+
+const { LDAP_CONFIG, LDAP_CRED, LDAP_USERS_FILE_PATH } = require('../constants');
+
+/**
+ * Import LDAP users to file with GUID attributes
+ * TODO: turn ldap requests into async and await functions
+ */
+const importLdapUsers = async () => {
+  try {
+    /* Read users.json file */
+    const users = JSON.parse(
+      fs.readFileSync(LDAP_USERS_FILE_PATH, { encoding: 'utf-8' }, (err) => {
+        throw Error(err);
+      }),
+    );
+
+    const usersWithoutGuids = users.filter(
+      (user) => !user.attributes && !user.attributes.idir_user_guid,
+    );
+
+    console.log(`Found ${users.length} users with ${usersWithoutGuids.length} missing guids`);
+
+    /* Create LDAP search filter */
+    const filter = new ldap.OrFilter({
+      filters: usersWithoutGuids.map((user) => {
+        console.log(user.username);
+        return new ldap.EqualityFilter({
+          attribute: 'samAccountName',
+          value: user.username,
+        });
+      }),
+    });
+
+    // create ldap client:
+    console.log('create ldap client');
+    const client = ldap.createClient({
```

Comment:
is this a sync or async function?

Reply:
it's sync originally, plan to turn it to async. but I'm waiting for some ldap credential to test it out first (see TODO comments). btw this is based on https://github.com/BCDevOps/keycloak-admin/pull/3